src/telemetry/downlink/downlink.c

```diff
@@ -1,14 +1,85 @@
 /* -----------------------------------------------------------------------------
  * Component Name: Downlink
  * Parent Component: Telemetry
- * Author(s): 
+ * Author(s): Adam Smialek
  * Purpose: Read telemetry messages from the queue and send them over the
  *          downlink whenever possible.
  * -----------------------------------------------------------------------------
  */
 
+#include <stdio.h>
+#include <string.h> // strlen()
+#include <arpa/inet.h> // inet_pton()
+#include <sys/socket.h>
+#include <unistd.h>
+
 #include "global_utils.h"
 
-int init_downlink( void ){
+static int udp_socket_;
+static struct sockaddr_in UDP_SERVER;
+
+/* prototypes declaration */
+int init_upd_socket(void);
+
+int init_downlink(void) {
+    int ret = init_upd_socket();
+    if (ret != SUCCESS) {
+        return ret;
+    }
+    return SUCCESS;
+}
+
+/**
+ * Sets up the socket on `SERVER_PORT` to work with UDP protocol.
+ *
+ * I have decided not to put this in `init_downlink()` for the function name to stay
+ * as much related to wwhat happens inside as possible and to make it more abstract.
+ * Just a quick digression and a heads up for further "standards", for anyone who
+ * reads these comments.
+ *
+ * @return
+ */
+int init_upd_socket(void) {
```

Comment:
It was intended that the `e_link` module should handle the ip communications. Note that the declarations in `e_link.h` are preliminary and can be changed. Currently, the contents of this file do not match the purpose of the module.